### PR TITLE
Update RFC-0140

### PR DESCRIPTION
**Pull request by philipr-za**
## Description
This PR updates RFC-0140 to better reflect the current implementation.

## How Has This Been Tested?
Only documentation updates.

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I have squashed my commits into a single commit.

### CODE REVIEW

File: RFC/src/Glossary.md [hunk 1]
```diff
@@ -86,6 +86,13 @@ kernels and headers up to that block height from the genesis block.
 A strategy for propagating messages amongst nodes in a peer-to-peer network. Example implementations of
 `BroadcastStrategy` include the Gossip protocol and flood fill.
 
+## Chain Reorganization
+[reorg]: #chain-reorg "After a chain split occurs nodes on the poorer chain must resync to the best chain"
+
+A chain reorganization occurs after a chain split occurs on the network, which commonly occurs due to network latency
+and connectivity issues. When a chain split occurs one chain will have the higher accumulated proof-of-work, this chain
+is considered the best chain. Nodes on the poorer chain will need to resync their chains to best chain. In this process
```
Comment (SWvheerden) [inline RFC/src/Glossary.md]:
```suggestion
is considered the best chain. Nodes on the poorer chain will need to rewind and resync their chains to best chain. In this process
```

File: RFC/src/RFC-0140_Syncing_and_seeding.md [hunk 2]
```diff
@@ -60,83 +60,93 @@ The aim of this Request for Comment (RFC) is to describe the syncing, seeding an
...
-divided into two [SynchronizationStrategy]s: complete sync and sync. Complete sync means that the node communicates with 
-an archive node to get the complete history of every single block from genesis block. Sync involves the node getting 
-every block from its [pruning horizon] to [current head], as well as every block header 
-from genesis block. 
+divided into two [SynchronizationStrategy]s: complete sync and horizon sync. Complete sync means that the node 
+communicates with an archive node to get the complete history of every single block from genesis block. Horizon Sync 
+involves the node getting every block from its [pruning horizon] to [current head], as well as every block header 
+up to the genesis block. 
+
+To determine if the node needs to synchronise the node will monitor the broadcast chain_metadata provided by its neighbours.
```
Comment (SWvheerden) [inline RFC/src/RFC-0140_Syncing_and_seeding.md:68]:
```suggestion
To determine if the node needs to synchronise, the node will monitor the broadcasted chain_metadata provided by it's neighbours.
```

File: RFC/src/RFC-0140_Syncing_and_seeding.md [hunk 3]
```diff
@@ -60,83 +60,93 @@ The aim of this Request for Comment (RFC) is to describe the syncing, seeding an
...
+1. Set [SynchronizationState] to `header_sync`.
+2. Sync all missing headers from the genesis block to the current chain tip. The initial header sync allows the node to
+   confirm that the syncing peer does indeed have a fully intact chain from which to sync that adheres to this nodes
+   consensus rules and has a valid proof-of-work that is higher than any competing chains.
+3. Set [SynchronizationState] to `horizon_sync`.
+4. Download all kernels from the current network tip back to this node's [pruning horizon].
+5. Validate kernel MMR root against headers.
+6. Download all [utxo]'s from the current network tip back to this node's [pruning horizon].
+7. Validate outputs and [utxo] MMR.
+8. Once all kernels and [utxo]s have been downloaded from the network tip back to this node's [pruning horizon] set
```
Comment (SWvheerden) [inline RFC/src/RFC-0140_Syncing_and_seeding.md]: Step 8  should be validate balance by comparing the utxo set to the total accumulated balance.
Step 9 should be do block sync to get to the listing state. 

horizon sync, only syncs to the horizon before handing over to normal block sync, it is also the only piece of code currently to check the total chain emmision

**Review by SWvheerden — DISMISSED** [review]
Looks good, one or 2 small nits

File: RFC/src/RFC-0140_Syncing_and_seeding.md (hunk 2 above)
Comment (philipr-za) [reply RFC/src/RFC-0140_Syncing_and_seeding.md:68]: nope

File: RFC/src/Glossary.md (hunk 1 above)
Comment (philipr-za) [reply RFC/src/Glossary.md]: updated

File: RFC/src/RFC-0140_Syncing_and_seeding.md (hunk 3 above)
Comment (philipr-za) [reply RFC/src/RFC-0140_Syncing_and_seeding.md]: Ok updated, but just to confirm you are saying that only the horizon sync checks the total chain emission. Why isn't that done with normal block sync?

Comment (SWvheerden) [reply RFC/src/RFC-0140_Syncing_and_seeding.md]: Its done indirectly. 
So normal block propagation only checks that that single block has the correct emission for its height. 
It checks that:
`outputs - inputs = emission of that height. `
So it only checks that the block has increased the total emission by the emission value of that height. If you do this for every block, you can assume that the total is also correct. 

Horizon sync does not check every block, so it needs to check that:
`sum(all_ouputs)  = total accumulated emission at that height.`
It has no blocks to check that the block emission was correct.

**Review by stringhandler — APPROVED** [review]

**Pull request merged**